### PR TITLE
Add Tesla Tank

### PR DESCRIPTION
**Pull request by MustaphaTR**
Note: I added `tstimpct` anim for this but Tesla Coil and Tesla Trooper is still not using it.

### CODE REVIEW

File: mods/ra2/rules/soviet-vehicles.yaml [hunk 1]
```diff
@@ -254,6 +254,54 @@ apoc:
 	ProducibleWithLevel:
 		Prerequisites: warfactory.infiltrated
 
+ttnk:
+	Inherits: ^Vehicle
+	Valued:
+		Cost: 1200
+	Tooltip:
+		Name: Tesla Tank
+		Description: Soviet Russian tank armed with dual small Tesla Coils.\n  Strong vs Vehicles, Infantry\n  Weak vs Aircraft
```
Comment (abcdefg30) [inline mods/ra2/rules/soviet-vehicles.yaml]: `Soviet Russian`? I think that is kinda duplicating... How about just `Russian`?

Comment (GraionDilach) [inline mods/ra2/rules/soviet-vehicles.yaml]: Russian special tank imo.

Comment (MustaphaTR) [inline mods/ra2/rules/soviet-vehicles.yaml]: Russia and Soviet Russia(Russian SFSR) is not same thing.

Comment (abcdefg30) [inline mods/ra2/rules/soviet-vehicles.yaml]: I just meant, as the faction name is also `Russia` and not `Russian SFSR`.

Comment (MustaphaTR) [inline mods/ra2/rules/soviet-vehicles.yaml]: OK i'm changing it.

File: mods/ra2/rules/soviet-vehicles.yaml [hunk 2]
```diff
@@ -254,6 +254,54 @@ apoc:
...
+	Tooltip:
+		Name: Tesla Tank
+		Description: Russian special tank armed with dual small Tesla Coils.\n  Strong vs Vehicles, Infantry\n  Weak vs Aircraft
+	Buildable:
+		Queue: Vehicle
+		BuildPaletteOrder: 50
+		Prerequisites: ~naweap, naradr, ~vehicles.russia
+	Mobile:
+		Speed: 75
+		ROT: 5
```
Comment (abcdefg30) [inline mods/ra2/rules/soviet-vehicles.yaml]: I think this is already the default value on `^Vehicle`, isn't it?

**Comment by abcdefg30** [comment]
We could polish the weapon a bit more I guess, but :+1: anyway.

File: mods/ra2/rules/soviet-vehicles.yaml (hunk 2 above)
Comment (phrohdoh) [inline mods/ra2/rules/soviet-vehicles.yaml]: Yes, it is.
@MustaphaTR please remove this.

**Comment by phrohdoh** [comment]
This weapon definitely needs more tweaking.
I don't see the left coil being used at all and the bolts appear to start quite a distance in front of the coils.

**Comment by GraionDilach** [comment]
The bolt should start between the coils. Prolly ClassicP*Fudge is needed to fix the FLH.

**Comment by penev92** [comment]
Between?  o.O

**Comment by GraionDilach** [comment]
Yes, between. The thing had no burst and WW decided to go that way.

**Comment by penev92** [comment]
Sure, but we can to it better than that.

**Comment by GraionDilach** [comment]
Well, in that scenario the D2k missile tower is the way to go.

**Comment by MustaphaTR** [comment]
Updated.

**Comment by MustaphaTR** [comment]
I tried to do same thing with D2k missile tower. But it doesn't work, it looks like it need `Burst:` to work.

Edit: Whatever i'll try to use Burst:.

**Comment by MustaphaTR** [comment]
Updated.

**Comment by MustaphaTR** [comment]
Updated again.

**Comment by phrohdoh** [comment]
``` crash
Exception of type `System.InvalidOperationException`: Actor does not have trait of type `OpenRA.Mods.Common.Traits.WarheadDebugOverlay`
  at OpenRA.TraitDictionary+TraitContainer`1[OpenRA.Mods.Common.Traits.WarheadDebugOverlay].Get (UInt32 actor) [0x0003e] in /Users/thill/projects/openra/OpenRA.Game//TraitDictionary.cs:154
  at OpenRA.TraitDictionary.Get[WarheadDebugOverlay] (OpenRA.Actor actor) [0x00013] in /Users/thill/projects/openra/OpenRA.Game//TraitDictionary.cs:90
  at OpenRA.Actor.Trait[WarheadDebugOverlay] () [0x0000d] in /Users/thill/projects/openra/OpenRA.Game//Actor.cs:217
  at OpenRA.Mods.Common.Warheads.SpreadDamageWarhead.DoImpact (WPos pos, OpenRA.Actor firedBy, IEnumerable`1 damageModifiers) [0x0003c] in /Users/thill/projects/openra/OpenRA.Mods.Common//Warheads/SpreadDamageWarhead.cs:56
  at OpenRA.Mods.Common.Warheads.DamageWarhead.DoImpact (Target target, OpenRA.Actor firedBy, IEnumerable`1 damageModifiers) [0x00038] in /Users/thill/projects/openra/OpenRA.Mods.Common//Warheads/DamageWarhead.cs:61
  at OpenRA.GameRules.WeaponInfo+<Impact>c__AnonStorey2.<>m__0 () [0x00027] in /Users/thill/projects/openra/OpenRA.Game//GameRules/WeaponInfo.cs:158
  at OpenRA.GameRules.WeaponInfo.Impact (Target target, OpenRA.Actor firedBy, IEnumerable`1 damageModifiers) [0x00095] in /Users/thill/projects/openra/OpenRA.Game//GameRules/WeaponInfo.cs:162
  at OpenRA.Mods.RA.Effects.TeslaZap.Tick (OpenRA.World world) [0x0009e] in /Users/thill/projects/openra/OpenRA.Mods.RA//Effects/TeslaZap.cs:58
  at OpenRA.World.<Tick>m__4 (IEffect e) [0x00002] in /Users/thill/projects/openra/OpenRA.Game//World.cs:289
  at OpenRA.WorldUtils.DoTimed[IEffect] (IEnumerable`1 e, System.Action`1 a, System.String text) [0x00022] in /Users/thill/projects/openra/OpenRA.Game//WorldUtils.cs:53
  at OpenRA.World.Tick () [0x0015e] in /Users/thill/projects/openra/OpenRA.Game//World.cs:289
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x001c4] in /Users/thill/projects/openra/OpenRA.Game//Game.cs:503
  at OpenRA.Game.LogicTick () [0x00055] in /Users/thill/projects/openra/OpenRA.Game//Game.cs:527
  at OpenRA.Game.Loop () [0x000d4] in /Users/thill/projects/openra/OpenRA.Game//Game.cs:651
  at OpenRA.Game.Run () [0x00042] in /Users/thill/projects/openra/OpenRA.Game//Game.cs:691
  at OpenRA.Program.Run (System.String[] args) [0x00011] in /Users/thill/projects/openra/OpenRA.Game//Support/Program.cs:117
  at OpenRA.Program.Main (System.String[] args) [0x00050] in /Users/thill/projects/openra/OpenRA.Game//Support/Program.cs:39
```

I will fix this.

``` diff
diff --git a/mods/ra2/rules/world.yaml b/mods/ra2/rules/world.yaml
index cb79e4e..ce6ff68 100644
--- a/mods/ra2/rules/world.yaml
+++ b/mods/ra2/rules/world.yaml
@@ -128,6 +128,7 @@ World:
        ResourceLayer:
        ResourceClaimLayer:
        PathfinderDebugOverlay:
+       WarheadDebugOverlay:
        SpawnMapActors:
        CreateMPPlayers:
        MPStartUnits@amcv:
```

**Comment by phrohdoh** [comment]
I rebased this fork and fixed some issues so had to merge this manually locally, but it has been included!

**Pull request closed without merging**